protos/flyteidl/event/event.proto

```diff
@@ -150,4 +150,20 @@ message TaskExecutionEvent {
     // that should be recorded regardless of the lack of phase change.
     // The version field should be incremented when metadata changes across the duration of an individual phase.
     uint32 phase_version                  = 12;    
+
+    // Metadata around how a task was executed.
+    TaskExecutionMetadata metadata        = 16;
+}
+
+// Holds metadata around how a task was executed.
+message TaskExecutionMetadata {
```

Comment:
This is great, can we conform to - https://github.com/lyft/flyte/issues/325

Reply:
hm I can add a freeform map as well? is there a specific reason this shouldn't have any structure?